airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -102,66 +140,66 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
             return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
         return {self.cursor_field: latest_benchmark}
 
-    def _create_prepared_request(
-        self, path: str, headers: Mapping = None, params: Mapping = None, json: Any = None, data: Any = None
-    ) -> requests.PreparedRequest:
-        """
-        Override to prepare request for AWS API.
-        AWS signature flow require prepared request to correctly generate `authorization` header.
-        Add `auth` arg to sign all the requests with AWS signature.
-        """
 
-        return self._session.prepare_request(
-            requests.Request(method=self.http_method, url=self.url_base + path, headers=headers, params=params, auth=self._aws_signature)
+class ReportsAmazonSPStream(AmazonSPStream, ABC):
+    primary_key = "reportId"
+    path_prefix = f"/reports/{REPORTS_API_VERSION}/reports"
+    wait_seconds = 30
+
+    def should_retry(self, response: requests.Response) -> bool:
+        should_retry = super().should_retry(response)
+        if not should_retry:
+            should_retry = response.json().get(self.data_field, {}).get("processingStatus") in ["IN_QUEUE", "IN_PROGRESS"]
+        return should_retry
+
+    def backoff_time(self, response: requests.Response):
```

Comment:
I get your intention but abusing retry mechanism for failures is not very nice, as we will have strange logs (see user_defined_backoff_handler and default_backoff_handler implementation)

Reply:
Leaving it as it is, as discussed in Slack